bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/OpenhabGraalJSScriptEngine.java

```diff
@@ -230,18 +230,20 @@ protected void beforeInvocation() {
         } catch (ScriptException e) {
             LOGGER.error("Could not inject global script", e);
         }
+
+        super.beforeInvocation();
```

Comment:
Normally you invoke the super method first, not last, if you don't care about the ordering.

I don't think that it's a big deal whether it should happen first or last, but it should probably be consistent (i.e. opposite with the invocation afterwards). I would advise that the superclass calls either both happen in a locked state, or both in an unlocked state. It looks like currently the `before` happens in a locked state, whereas the `after` occurs in an unlocked state. I would probably choose to have both occur in an unlocked state.